src/content/helpers/player-profile.js

```diff
@@ -4,6 +4,13 @@ import { getCurrentPath } from './location'
 export const getPlayerProfileNickname = path => {
   const match = /players(?:-modal)?\/([^/]+)\//.exec(path || getCurrentPath())
 
+  if (!match || !match[1]) {
+    const profile = /players(?:-modal)?\/([^/]+)$/.exec(
+      path || getCurrentPath()
+    )
+    return profile && profile[1]
+  }
+
```

Comment:
Why is this required and how is this different from the `.exec` above?

Reply:
Lets say you are currently in path `/en/players/username`. Current upstream code returns null with `getPlayerProfileNickname()` method. This method worked fine with `/en/players//stats/csgo`

Reply:
Okay so lets say you are in your profile: `/en/players/username`. This current upstream method `getPlayerProfileNickname` does not work with this path and only works when client navigates one of the sub-routes of the profile eg. `/en/players/username/stats/csgo` and since we want to show the badge always on profile, not on sub-routes only, this is required. 

This is 'dirty' batch and could be handled with better regex query. I couldn't figure out a working regex query to this problem.

Reply:
The `master` regex detects the nickname for all player profile paths: https://regex101.com/r/UzU8v4/1

I've also tested your branch in the browser and the badge is added on all player profile links.

Reply:
> I've also tested your branch in the browser and the badge is added on all player profile links.

My branch does this minor trick to get it working but see the images:

Path: `/en/players/Morabotti`
![matchingProblem](https://user-images.githubusercontent.com/42501026/80478605-6ac96e00-8956-11ea-8b0a-15f3fa04c10e.png)

Path: `/en/players/Morabotti/stats/csgo`
![matchingProblem2](https://user-images.githubusercontent.com/42501026/80478650-7a48b700-8956-11ea-88e2-df66badb9514.png)

As you can see, for some reason the first match does return null even though its exactly same regex query as masters. I've working regex query that returns correct name for both of paths, is this good?:

![possibleFix](https://user-images.githubusercontent.com/42501026/80479867-62723280-8958-11ea-8e20-facb029ff229.png)
![possibleFix2](https://user-images.githubusercontent.com/42501026/80479878-6605b980-8958-11ea-9a53-905970eff99e.png)

Reply:
Try this regex and the "hack" can be removed. Worked for me.

```diff
diff --git a/src/content/helpers/player-profile.js b/src/content/helpers/player-profile.js
index 2c0f7a0..d26d868 100644
--- a/src/content/helpers/player-profile.js
+++ b/src/content/helpers/player-profile.js
@@ -2,14 +2,7 @@
 import { getCurrentPath } from './location'

 export const getPlayerProfileNickname = path => {
-  const match = /players(?:-modal)?\/([A-Za-z0-9_-]+)/.exec(path || getCurrentPath())
-
-  if (!match || !match[1]) {
-    const profile = /players(?:-modal)?\/([^/]+)$/.exec(
-      path || getCurrentPath()
-    )
-    return profile && profile[1]
-  }
+  const match = /players(?:-modal)?\/([A-Za-z0-9_-]+)(?:\/)?/.exec(path || getCurrentPath())

   return match && match[1]
 }
```